From 15c75833378d943946161c6b2b3c9074b5c8d30f Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Sun, 24 Sep 2017 22:43:16 +0100
Subject: [PATCH] vc-sm: Fix a printf of a dma_addr_t from %u to %pad

Avoids issues when other build parameters result in
dma_addr_t being a 64 bit value.
See #2196.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/char/broadcom/vc_sm/vmcs_sm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/char/broadcom/vc_sm/vmcs_sm.c b/drivers/char/broadcom/vc_sm/vmcs_sm.c
index 034ae2f27f870..e7ec639937974 100644
--- a/drivers/char/broadcom/vc_sm/vmcs_sm.c
+++ b/drivers/char/broadcom/vc_sm/vmcs_sm.c
@@ -573,8 +573,8 @@ static int vc_sm_global_state_show(struct seq_file *s, void *v)
 				   resource->attach);
 			seq_printf(s, "           SGT          %p\n",
 				   resource->sgt);
-			seq_printf(s, "           DMA_ADDR     0x%08X\n",
-				   resource->dma_addr);
+			seq_printf(s, "           DMA_ADDR     %pad\n",
+				   &resource->dma_addr);
 		}
 	}
 	seq_printf(s, "\n\nTotal resource count:   %d\n\n", resource_count);
